src/main.rs

```diff
@@ -147,35 +230,126 @@ fn main() {
                     MouseMode::CameraLook => renderer.camera.move_on_orbit(mouse_delta),
                     MouseMode::CameraPan => renderer.camera.move_focus(mouse_delta),
                 }
+                ui_state.camera_target[0] = renderer.camera.target[0];
+                ui_state.camera_target[1] = renderer.camera.target[1];
+                ui_state.camera_target[2] = renderer.camera.target[2];
+                ui_state.camera_azimuth = renderer.camera.azimuth;
+                ui_state.camera_elevation = renderer.camera.elevation;
                 prev_mouse = position;
             }
             Event::MainEventsCleared => {
                 window.request_redraw();
             }
             Event::RedrawRequested(_) => {
                 // Redraw the application.
+                let frame_time_delta = prev_frame_time.elapsed();
+                prev_frame_time = imgui_context.io_mut().update_delta_time(prev_frame_time);
+
                 let frame = renderer
                     .swap_chain
                     .get_next_texture()
                     .expect("Timeout when acquiring next swap chain texture");
                 let mut commands = renderer
                     .device
                     .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
-                renderer.render(&mut commands, &frame.view, &rendering::defaults::axes(), &rendering::defaults::render_all_vertices(&rendering::defaults::axes()), true);
-                //renderer.render(&mut commands, &frame.view, &vertices, &indecies, false);
+                renderer.render(
+                    &mut commands,
+                    &frame.view,
+                    &rendering::defaults::axes(),
+                    &rendering::defaults::render_all_vertices(&rendering::defaults::axes()),
+                    true,
+                );
                 for i in 0..lines.len() {
                     let v = &lines[i].verticies;
                     let i = &lines[i].indicies;
                     renderer.render(&mut commands, &frame.view, v, &i, false);
                 }
                 
-                renderer.queue.submit(&[commands.finish()]);
+                let mut imgui_commands: wgpu::CommandEncoder = renderer
+                    .device
+                    .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+
+                if ui_state.gui_open {
+
+                platform
+                    .prepare_frame(imgui_context.io_mut(), &window)
+                    .expect("Failed to prepare frame");
+                let ui = imgui_context.frame();
+
+                {
+                    let window = imgui::Window::new(im_str!("Display"));
```

Comment:
So as part of the refactor it'd be nice if you made things into systems.

For instance the thing that does the orbit cam, can be a system that registers, and is also in charge of it's imgui render window.  That way it's all in roughly the same place.

Unless you are intentionally opting for a lower file count, then I would recommend one file for your imgui controls, one for your common/state data types, etc...